common/djangoapps/student/models.py

```diff
@@ -1707,8 +1708,9 @@ def upgrade_deadline(self):
             )
             return None
 
+        schedule_config = CourseScheduleConfiguration.current(self.course_id)
         try:
-            if self.schedule:
+            if schedule_config.enabled and schedule_config.verified_upgrade_deadline_enabled and self.schedule:
```

Comment:
My concern was that we likely want to operate in a mode where schedules are being created but aren't being used to display a verified upgrade deadline.

We need some kind of flag that allows us to enable creation without immediately overriding the deadline.